stmt/stmts/update.go

```diff
@@ -82,37 +85,27 @@ func (s *UpdateStmt) SetText(text string) {
 	s.Text = text
 }
 
-func getUpdateColumns(t table.Table, assignList []expressions.Assignment) ([]*column.Col, error) {
-	tcols := make([]*column.Col, len(assignList))
-	for i, asgn := range assignList {
+func getUpdateColumns(t table.Table, assignList []expressions.Assignment, multipleTable bool) ([]*column.Col, error) {
+	tcols := make([]*column.Col, 0, len(assignList))
+	tname := t.TableName()
+	for _, asgn := range assignList {
+		if multipleTable {
+			if !strings.EqualFold(tname.O, asgn.TableName) {
+				continue
+			}
+		}
 		col := column.FindCol(t.Cols(), asgn.ColName)
 		if col == nil {
+			if multipleTable {
```

Comment:
In my impression, we only check columns validity in `select fields`, `where`, `groupby`, `having` and `orderby`, perhaps not `join`, i am not sure, it is better to have a try.